network/server/tchannelthrift/node/service.go

```diff
@@ -277,3 +278,8 @@ func (s *service) WriteBatch(tctx thrift.Context, req *rpc.WriteBatchRequest) er
 	}
 	return nil
 }
+
+func (s *service) TruncateNamespace(tctx thrift.Context, req *rpc.TruncateNamespaceRequest) error {
+	s.db.TruncateNamespace(req.Ns)
+	return nil
```

Comment:
Should just be able to grab a RLock and take a snap of the `len(s.lookup)` and add them across the current owned shards perhaps before calling `initShards()` in namespace?